rollup/internal/controller/watcher/batch_proposer.go

```diff
@@ -29,12 +29,8 @@ type BatchProposer struct {
 	chunkOrm   *orm.Chunk
 	l2BlockOrm *orm.L2Block
 
-	maxL1CommitGasPerBatch          uint64
-	maxL1CommitCalldataSizePerBatch uint64
-	batchTimeoutSec                 uint64
-	gasCostIncreaseMultiplier       float64
-	maxUncompressedBatchBytesSize   uint64
-	maxChunksPerBatch               int
+	batchTimeoutSec   uint64
```

Comment:
Why don't we use the `cfg *config.BatchProposerConfig` directly? This would make future adjustments of config parameters easier (and less error prone) as we don't need to pass things around so much. Same applies to `ChunkProposer` and `BundleProposer`

Reply:
Nice catch. Changed based on the comment in [1e52cc2](https://github.com/scroll-tech/scroll/pull/1658/commits/1e52cc206d8ee05010a724d59999deba54112cf2).